uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/utils/PathUtil.java

```diff
@@ -37,6 +40,21 @@
         return Paths.convert(path);
     }
 
+    public org.uberfire.backend.vfs.Path convert(org.uberfire.java.nio.file.Path path) {
+        return Paths.convert(path);
+    }
+
+    public String stripProtocolAndBranch(String uri) {
+        return uri.replaceFirst("^[A-Za-z]+://([^@]+@)?", "");
```

Comment:
It would be nice to have these new methods unit tested.
Also `String#replaceFirst` is calling `Pattern.compile` internally. Based on how often we're expected to call this method, wouldn't it make sense to pre-compile the pattern in private static final field of this class?